src/tendril.rs

```diff
@@ -68,30 +74,56 @@ pub unsafe trait Atomicity: 'static {
 /// and so doesn't typically need to be written.
 ///
 /// This is akin to using `Rc` for reference counting.
-pub struct NonAtomic(Cell<usize>);
+#[repr(C, align(4))]
+pub struct NonAtomic(Cell<NonAtomicInternals>);
```

Comment:
Wouldn’t a struct of separate cells be more convenient that a cell of struct?

Reply:
Probably, yes. This is a leftover from when the the entire structure was packed, but only the reference counter needs to be packed now.